codegen/lib/graphql_swift_gen/templates/type.swift.erb

```diff
@@ -214,36 +214,41 @@ extension <%= schema_name %> {
   <% when 'INPUT_OBJECT' %>
     open class <%= type.name %> {
       <% type.required_input_fields.each do |field| %>
-       open var <%= escape_reserved_word(field.camelize_name) %>: <%= swift_input_type(field.type) %>
+
+       open var <%= escape_reserved_word(field.camelize_name) %>: <%= swift_input_type(field.type, non_null: true) %>
       <% end %>
       <% type.optional_input_fields.each do |field| %>
+
         open var <%= escape_reserved_word(field.camelize_name) %>: <%= swift_input_type(field.type) %> {
           didSet {
-            <%= escape_reserved_word(field.camelize_name) %>Seen = true
+            serialize<%= escape_reserved_word(field.classify_name) %> = true
           }
         }
-        private var <%= escape_reserved_word(field.camelize_name) %>Seen: Bool = false
+        // Observes if the <%= escape_reserved_word(field.camelize_name) %> property was set, and indicates whether it should be serialized.
+        open var serialize<%= escape_reserved_word(field.classify_name) %>: Bool
       <% end %>
 
       public init(
         <% input_fields = type.required_input_fields + type.optional_input_fields # manually ordered with required fields first %>
         <% input_fields.each do |field| %>
           <% seperator = field == input_fields.last ? "" : "," %>
           <% if field.type.non_null? %>
-            <%= escape_reserved_word(field.camelize_name) %>: <%= swift_input_type(field.type) %><%= seperator %>
+            <%= escape_reserved_word(field.camelize_name) %>: <%= swift_input_type(field.type, non_null: true) %><%= seperator %>
           <% else %>
-            <%= escape_reserved_word(field.camelize_name) %>: <%= swift_input_type(field.type) %>? = nil<%= seperator %>
+            <%= escape_reserved_word(field.camelize_name) %>: <%= swift_input_type(field.type) %> = nil,
+            serialize<%= escape_reserved_word(field.classify_name) %>: Bool = false<%= seperator %>
           <% end %>
         <% end %>
       ) {
         <% type.required_input_fields.each do |field| %>
           self.<%= escape_reserved_word(field.camelize_name) %> = <%= escape_reserved_word(field.camelize_name) %>
         <% end %>
         <% type.optional_input_fields.each do |field| %>
+          self.serialize<%= escape_reserved_word(field.classify_name) %> = serialize<%= escape_reserved_word(field.classify_name) %>
           <% field_name = escape_reserved_word(field.camelize_name) %>
           if let <%= field_name %> = <%= field_name %> {
-            self.<%= field_name %>Seen = true
             self.<%= field_name %> = <%= field_name %>
+            self.serialize<%= escape_reserved_word(field.classify_name) %> = true
```

Comment:
Doesn't the `self.<%= field_name %> = <%= field_name %>` line already set the serialize value to `true` in the properties `didSet`

Reply:
No, `didSet` observers arent triggered in initializers